File: week1/cash/cash.c
#include <cs50.h>
#include <math.h>
#include <stdio.h>

int main(void)
{
    int change = 0;
    float dollars;

    do
    {
         dollars = get_float("Change owed: ");
    }
    while (dollars <= 0);

    int cents = round(dollars * 100);

    while (cents > 0)
    {
        if (cents >= 25)
        {
            change ++;
            cents = cents - 25;
        }
        else if (cents >= 10)
        {
            change ++;
            cents = cents - 10;
        }
        else if (cents >= 5)
        {
            change ++;
            cents = cents - 5;
        }
        else
        {
            change += cents;
            cents = 0;
        }
    }

    printf("%d\n", change);
}

File: week6/cash/cash.py
from cs50 import get_float


def main():
    while True:
        dollars = get_float("Change owed: ")

        if dollars > 0:
            break

    cents = round(dollars * 100)
    change = 0

    while cents > 0:
        if cents >= 25:
            change += 1
            cents -= 25
        elif cents >= 10:
            change += 1
            cents -= 10
        elif cents >= 5:
            change += 1
            cents -= 5
        else:
            change += cents
            cents = 0

    print(change)


main()